### ISSUE TITLE
Investigate opening space after brace in certain scenarios

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1234]
*Single line functions*

    public string filepath {get; set; }

    bool HasPending() {return m_HasPendingDeletes != 0; }

    public AnimationCurve curveToSaveAsPreset {get {return m_Curve; } set {m_Curve = value; }}

This seems to happen with no spaces on the interior. Opening brace doesn't get one, closing brace does.

Turning on sp_inside_braces=add fixes it, but also changes a lot of initializer code we don't want to touch (like x = {1}). May need special support, or perhaps there's a bug..

*Initializers*

Not sure if this is what we want..

    session.CustomEvent("customEvent1", new Dictionary<string, object>(){ { "k1", "v1" }, { "k2", "v2" } });
    ... --> ...
    session.CustomEvent("customEvent1", new Dictionary<string, object>() { { "k1", "v1" }, { "k2", "v2" } });

Second line adds a space before the initializer {. Is that what we want for C#?

Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\10051-UNI-1338.cs" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\10051-UNI-1338.cs"
index 3bdb6f0..e6f6642 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\10051-UNI-1338.cs"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\10051-UNI-1338.cs"
@@ -1,27 +1,27 @@
 // *Single line functions*
 
 public class Class
 {
     public string foo {get; set;}
 
     bool HasBar() {return m_HasBar != 0;}
 
     public Bar prop {get {return m_bar;} set {m_bar = value;}}
 
 // This seems to happen with no spaces on the interior. Opening brace doesn't get one, closing brace does.
 
 // Turning on sp_inside_braces=add fixes it, but also changes a lot of initializer code we don't want to touch (like x = {1}). May need special support, or perhaps there's a bug..
 
 // *Initializers*
 
 // Not sure if this is what we want..
 
     public void foo()
     {
-        sas.Foo("bar", new Dictionary<string, object>(){ { "k1", "v1" }, { "k2", "v2" } });
+        sas.Foo("bar", new Dictionary<string, object>() { { "k1", "v1" }, { "k2", "v2" } });
         // ... --> ...
         sas.Foo("bar", new Dictionary<string, object>() { { "k1", "v1" }, { "k2", "v2" } });
     }
 
 // Second line adds a space before the initializer {. Is that what we want for C#?
 }
```

Debug support files:
---
[10051-UNI-1338.cs](https://gist.githubusercontent.com/mihaipopescu/df0440587c57a3edddbe34d2dc7f85e5/raw/3bdb6f011d6817699a963a9a18dda4077f76c2b9/10051-UNI-1338.cs)
[10051-UNI-1338.cs.log](https://gist.githubusercontent.com/mihaipopescu/df0440587c57a3edddbe34d2dc7f85e5/raw/ce42ed221bcfbce95c960c2029fcbcb35c240b41/10051-UNI-1338.cs.log)
[10051-UNI-1338.cs.unc](https://gist.githubusercontent.com/mihaipopescu/df0440587c57a3edddbe34d2dc7f85e5/raw/857d2c417a23db55f1424a872c0720b755b2f603/10051-UNI-1338.cs.unc)
[Uncrustify.CSharp.cfg](https://gist.githubusercontent.com/mihaipopescu/df0440587c57a3edddbe34d2dc7f85e5/raw/1d064c1fabb76ce328dd015ed40023207de20ca1/Uncrustify.CSharp.cfg)
[UNI-1338.cs](https://gist.githubusercontent.com/mihaipopescu/df0440587c57a3edddbe34d2dc7f85e5/raw/e32991e51e812c4524426a7483337d1a28049f4b/UNI-1338.cs)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by CDanU** [comment]
Remove: 
```sql
# Add or remove space between ')' and '{' of function
sp_fparen_brace                 = add      # ignore/add/remove/force
```

**Comment by mihaipopescu** [comment]
Yes, indeed it that one! Using the reduced config with that option disabled the space is not added anymore. I'll run the new config against our codebase to check for possible unwanted regressions.

```ini
#sp_fparen_brace                 = add
indent_columns                  = 4
indent_with_tabs                = 0
indent_class                    = true
nl_class_leave_one_liners       = true
nl_end_of_file                  = force
nl_end_of_file_min              = 1
```

Unsure now if this issue is related to #197 where we had an issue with the fix made which caused other regressions. They seem to be fixed as seen that in this test only the last diff is still a problem.

**Comment by CDanU** [comment]
Even if that might fix the problem on your site, there is a reason why you run into that problem. 
The option should only affect __function__ parenthesis. The example you posted is a brace initializer right?

**Comment by mihaipopescu** [comment]
Yes. We probably need that option.

The constructor call can be tokenized as such and not a simple function call and probably then we can properly detect the initializer.

**Comment by gmaurel** [comment]
Fixed at 55cc441b4a343ca669a8dee52ac714a8a616f668,  Jan 12 18:49:47 2018
